### PR TITLE
Add NVM role

### PR DESCRIPTION
**Pull request by TomAdam**

### CODE REVIEW

File: nvm/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Installs NVM on a user by user basis. Can co-exist with a system Node version. 
+
+Config
+------
+
+```yaml
+nvm_config:
+- user: vagrant # username to install NVM for
+  default_version: lts # any version NVM understands to set as default
+  versions: # array of versions to install for user
```
Comment (lopsided) [inline nvm/README.md]: Is this not over-complicating things? Do you think a single user will need multiple versions on a provisioned vm?

File: nvm/tasks/per_user.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+---
+- become_user: '{{ nvm_user_config.user }}'
+  become: true
+  block:
+
+  - name: Check NVM version
+    command: 'bash -ic "nvm --version"'
+    failed_when: nvm_installed_version.rc != 0 and nvm_installed_version.rc != 127
```
Comment (lopsided) [inline nvm/tasks/per_user.yml]: What's with the 127? Does that need parametrising?

File: nvm/tasks/per_user.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+---
+- become_user: '{{ nvm_user_config.user }}'
+  become: true
+  block:
```
Comment (lopsided) [inline nvm/tasks/per_user.yml]: I can see the block below being conditional, but can't see why you have wrapped the rest in a block. Could you not have put the become_user and become into the loop task in main.yml above and put these as individual tasks? Is this just a style thing again?

File: nvm/README.md (hunk 1 above)
Comment (TomAdam) [reply nvm/README.md]: We have that case over at TT- the inspiration for this role over using a system node version.

File: nvm/tasks/per_user.yml (hunk 3 above)
Comment (TomAdam) [reply nvm/tasks/per_user.yml]: Become doesn't pass through when doing an include, so it has to be on a block like this. I think there may be a fix in the works to allow it to apply to the include.

File: nvm/tasks/per_user.yml (hunk 2 above)
Comment (TomAdam) [reply nvm/tasks/per_user.yml]: No idea ::/:

Comment (TomAdam) [reply nvm/tasks/per_user.yml]: 127 is the bash return code for command not found. I remember how this works now. If you say only `nvm_installed_version.rc != 127` then 0 is treated as a fail. I'll add a comment to make things clearer.

Comment (TomAdam) [reply nvm/tasks/per_user.yml]: Added in #0fa5a9b2fdbf627277fda11792484b7e24f7d928

**Pull request merged**